tests/utils/test_dist__gather__losses.py

```diff
@@ -0,0 +1,173 @@
+import torch
+import torch.distributed
+from pytorch_lightning import LightningModule, Trainer
+from pytorch_lightning.strategies.ddp import DDPStrategy
+from torch import Tensor
+from torch.nn import Linear, Module
+from torch.optim import SGD
+from torch.utils.data import DataLoader, TensorDataset
+from torchvision.datasets import FakeData
+from torchvision.transforms import ToTensor
+
+from lightly.data import LightlyDataset
+from lightly.loss.dcl_loss import DCLLoss
+from lightly.loss.ntx_ent_loss import NTXentLoss
+from lightly.loss.tico_loss import TiCoLoss
+from lightly.loss.vicreg_loss import VICRegLoss
+from tests.utils.benchmarking.test_benchmark_module import (
+    _DummyModel as _BenchmarkDummyModel,
+)
+
+"""
+WARNING: 
+Using a DDPStrategy causes the file to be executed once per device.
+Thus this test needs to be in a separate file.
+"""
+
+
+class Model(LightningModule):
+    def __init__(self, gather: bool, criterion: Module, learning_rate: float):
+        super().__init__()
+        self.model = Linear(5, 2, bias=False)
+        self.model.weight.data = torch.Tensor(
+            [
+                [0.1, 0.2, 0.3, 0.4, 0.5],
+                [0.6, 0.7, 0.8, 0.9, 1.0],
+            ]
+        )
+        self.criterion = criterion
+        self.gather = gather
+        self.learning_rate = learning_rate
+
+    def training_step(self, batch: Tuple[Tensor, Tensor], batch_idx: int) -> Tensor:
+        x = batch[0]
+        y = batch[1]
+        x = self.model(x).flatten(start_dim=1)
+        y = y.flatten(start_dim=1)
+        loss: Tensor = self.criterion(x, y)
+        return loss
+
+    def configure_optimizers(self) -> Any:
+        return SGD(self.parameters(), lr=self.learning_rate)
+
+
+@pytest.fixture
+def close_torch_distributed() -> Generator[None, None, None]:
+    yield None
+    torch.distributed.destroy_process_group()
+
+
+class TestGatherLayer_Losses:
+    """
+    Tests that the gather layer works as expected.
+
+    Because using a DDPStrategy causes the whole script to be executed multiple
+    times, running a proper test is difficult. The approach used here:
+
+    1. This test was run once with n_devices=1 and gather=False. The resulting
+    parameters after 10 epochs are hardcoced as expected_params__10_epochs__no_gather.
+
+    2. This test is now run with n_devices=2 and gather=True. The resulting
+    parameters are asserted to be the same ad with n_devices=1 and gather=False.
+
+    Note that the results would not be the same for n_devices=1 and n_devices=2 if
+    there was:
+    - Any randomness in the transform, as the order the samples are processed and
+    thus the random seed differ.
+    - Any batch normalization in the model, as the batch size is split between
+    the devices and there is no gather layer for the batch normalization.
+    """
+
+    def test_loss_ntxent(self, close_torch_distributed: None) -> None:
+        self._test_ddp(
+            NTXentLoss,
+            torch.Tensor(
```

Comment:
```suggestion
            torch.tensor(
```

`torch.Tensor` to create a new tensor is deprecated